### PR TITLE
8334758: Incorrect note in Javadoc for a few RandomGenerator methods

### PR DESCRIPTION
**Pull request by rgiulietti**
Small corrections to @implSpec notes in a few methods in RandomGenerator.

---------
### Progress
- [x] Change must be properly reviewed (1 review required, with at least 1 [Reviewer](https://openjdk.org/bylaws#reviewer))
- [x] Change requires CSR request [JDK-8336288](https://bugs.openjdk.org/browse/JDK-8336288) to be approved
- [x] Change must not contain extraneous whitespace
- [x] Commit message must refer to an issue

### Issues
 * [JDK-8334758](https://bugs.openjdk.org/browse/JDK-8334758): Incorrect note in Javadoc for a few RandomGenerator methods (**Bug** - P4)
 * [JDK-8336288](https://bugs.openjdk.org/browse/JDK-8336288): Incorrect note in Javadoc for a few RandomGenerator methods (**CSR**)

### Reviewers
 * [Brian Burkhalter](https://openjdk.org/census#bpb) (@bplb - **Reviewer**)

### Reviewing
<details><summary>Using <code>git</code></summary>

Checkout this PR locally: \
`$ git fetch https://git.openjdk.org/jdk.git pull/20152/head:pull/20152` \
`$ git checkout pull/20152`

Update a local copy of the PR: \
`$ git checkout pull/20152` \
`$ git pull https://git.openjdk.org/jdk.git pull/20152/head`

</details>
<details><summary>Using Skara CLI tools</summary>

Checkout this PR locally: \
`$ git pr checkout 20152`

View PR using the GUI difftool: \
`$ git pr show -t 20152`

</details>
<details><summary>Using diff file</summary>

Download this PR as a diff file: \
<a href="https://git.openjdk.org/jdk/pull/20152.diff">https://git.openjdk.org/jdk/pull/20152.diff</a>

</details>

### Webrev
[Link to Webrev Comment](https://git.openjdk.org/jdk/pull/20152#issuecomment-2224890425)

### CODE REVIEW

File: src/java.base/share/classes/java/util/random/RandomGenerator.java [hunk 1]
```diff
@@ -809,12 +809,11 @@ default int nextInt() {
...
-     * positive {@code int}. Then invokes {@code nextInt()}, limiting the result
-     * to be greater than or equal zero and less than {@code bound}. If {@code bound}
-     * is a power of two then limiting is a simple masking operation. Otherwise,
-     * the result is re-calculated by invoking {@code nextInt()} until the
-     * result is greater than or equal zero and less than {@code bound}.
+     * @implSpec The default implementation checks that {@code bound} is positive.
+     * It then invokes {@link #nextInt()} one or more times to ensure a uniform
+     * distribution in the range 0 (inclusive)
+     * to {@code bound} (exclusive).
+     * It assumes the distribution of {@link #nextInt()} to be uniform in turn.
```
Comment (bplb) [inline src/java.base/share/classes/java/util/random/RandomGenerator.java:816]: Do you need "in turn" here (and in the other changes)?

**Review by bplb — APPROVED** [review]
Looks fine but maybe a second Reviewer should agree?

**Comment by rgiulietti** [comment]
I guess that the approval of the CSR, once done, would "de-facto" count as a 2nd review? After all, this is a "CSR only" change.

**Comment by bplb** [comment]
> I guess that the approval of the CSR, once done, would "de-facto" count as a 2nd review?

I agree.

**Pull request closed without merging**